pubtools/_quay/iib_operations.py

```diff
@@ -0,0 +1,270 @@
+        if setting not in target_settings:
+            raise InvalidTargetSettings(
+                "'{0}' must be present in the target settings.".format(setting)
+            )
+
+    required_docker_settings = ["umb_urls", "docker_reference_registry"]
+    for setting in required_docker_settings:
+        if setting not in target_settings["docker_settings"]:
+            raise InvalidTargetSettings(
+                "'{0}' must be present in the docker settings.".format(setting)
+            )
+    if (
+        "iib_overwrite_from_index_token" in target_settings
+        and "iib_overwrite_from_index" not in target_settings
+    ) or (
+        "iib_overwrite_from_index_token" not in target_settings
+        and "iib_overwrite_from_index" in target_settings
+    ):
+        msg = (
+            "Either both or neither of 'iib_overwrite_from_index' and "
+            "'iib_overwrite_from_index_token' should be specified in target settings."
+        )
+        LOG.error(msg)
+        raise InvalidTargetSettings(msg)
+
+
+def task_iib_add_bundles(
+    bundles, archs, index_image, deprecation_list, signing_key, hub, task_id, target_settings
+):
+    """
+    Perform all the necessary actions for the 'PushAddIIBBundles' entrypoint.
+
+    Args:
+        bundles ([str]):
+            Bundles to add to the index image.
+        archs ([str]):
+            Architectures to build the index image for.
+        index_image (str):
+            Index image to add the bundles to.
+        deprecation_list ([str]):
+            Bundles to deprecate in the index image.
+        signing_key (str):
+            Signing key to be used.
+        hub (HubProxy):
+            Instance of XMLRPC pub-hub proxy.
+        task_id (str):
+            ID of the pub task.
+        target_settings (dict):
+            Dictionary containing settings necessary for performing the operation.
+    """
+    image_schema = "{host}/{namespace}/{repo}:{tag}"
+    verify_target_settings(target_settings)
+
+    # Build new index image in IIB
+    build_details = OperatorPusher.iib_add_bundles(
+        bundles=bundles,
+        archs=archs,
+        index_image=index_image,
+        deprecation_list=deprecation_list,
+        target_settings=target_settings,
+    )
+
+    _, tag = build_details.index_image.split(":", 1)
+    dest_image = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo=get_internal_container_repo_name(target_settings["quay_operator_repository"]),
+        tag=tag,
+    )
+    # TODO: does target_settings["quay_namespace"] point to rh-osbs?
+    # Index image used to fetch manifest list. This image will never be overwritten
+    intermediate_index_image = image_schema.format(
+        host=target_settings.get("quay_host", "quay.io").rstrip("/"),
+        namespace=target_settings["quay_namespace"],
+        repo="iib",
```

Comment:
This is not true for staging iib (it's iib-stage there). Emily pointed out in my code review in pub and it applies also here.

Reply:
Actually I was wrong, it's really iib. Only org is different per envs, which is correct here